src/test/mochitest/browser_dbg-search-project.js

```diff
@@ -0,0 +1,67 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+function toggleProjectSearch() {
+  synthesizeKeyShortcut("CmdOrCtrl+Shift+f");
+}
+
+function getResult(dbg) {
+  return findElementWithSelector(dbg, ".managed-tree .result");
+}
+
+async function selectResult(dbg) {
+  const item = getResult(dbg);
+  const select = waitForDispatch(dbg, "SELECT_SOURCE");
+  item.click();
+  return select;
+}
+
+function getResultsCount(dbg) {
+  const matches = dbg.selectors
+    .getTextSearchResults(dbg.getState())
+    .valueSeq()
+    .map(file => file.matches)
+    .toJS();
+
+  return [].concat(...matches).length;
```

Comment:
also, since we are doing toJS() at the end here, maybe it makes more sense to do matches.size